### PR TITLE
Eliminate nullness errors from sdks/java/extensions/ml

### LINKED ISSUE #20499: Eliminate nullability errors from :sdks:java:extensions:ml
Just stop suppressing nullness warnings in some classes and fix the errors!

Imported from Jira [BEAM-10531](https://issues.apache.org/jira/browse/BEAM-10531). Original Jira may contain additional context.
Reported by: jayendra13.
Subtask of issue #20497

### PR DESCRIPTION
**Pull request by kennknowles**
Fixes #20499

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [x] Update `CHANGES.md` with noteworthy changes.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by kennknowles** [comment]
run java precommit

**Comment by kennknowles** [comment]
Run Java PreCommit

**Comment by kennknowles** [comment]
Run Java PreCommit

**Comment by kennknowles** [comment]
run java precommit

**Comment by kennknowles** [comment]
run java precommit

**Comment by kennknowles** [comment]
Run Java PreCommit

**Comment by kennknowles** [comment]
More FlinkRequiresStableInputTest

**Comment by kennknowles** [comment]
Run Java PreCommit

**Comment by Abacn** [comment]
Possible related postcommit test failure: https://ci-beam.apache.org/job/beam_PostCommit_Java/10394/testReport/junit/org.apache.beam.sdk.extensions.ml/RecommendationAIUserEventIT/createUserEvent/

File: sdks/java/extensions/ml/src/main/java/org/apache/beam/sdk/extensions/ml/RecommendationAIWriteUserEvent.java [hunk 1]
```diff
@@ -60,13 +58,13 @@ static Builder newBuilder() {
   }
 
   /** @return ID of Google Cloud project to be used for creating user events. */
-  public abstract @Nullable String projectId();
```
Comment (Abacn) [inline sdks/java/extensions/ml/src/main/java/org/apache/beam/sdk/extensions/ml/RecommendationAIWriteUserEvent.java:63]: Due to the change here.

In general it is possible breakage could happen elsewhere forcing previously nullable parameters to nonnull

Comment (Abacn) [reply sdks/java/extensions/ml/src/main/java/org/apache/beam/sdk/extensions/ml/RecommendationAIWriteUserEvent.java:63]: btw was this PR unreviewed?

Comment (kennknowles) [reply sdks/java/extensions/ml/src/main/java/org/apache/beam/sdk/extensions/ml/RecommendationAIWriteUserEvent.java:63]: Oh sheesh. You are right. My mistake! I thought this was reviewed. I was kicking it for a long time on flakes.

**Comment by kennknowles** [comment]
Revert at #26620 both because it was not reviewed and because of post commit failure.

**Pull request merged**